src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2753,29 +2753,34 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     /** String representation of symbol's definition following its name */
     final def infoString(tp: Type): String = {
       def parents = (
-        if (settings.debug.value) parentsString(tp.parents)
+        if (settings.debug) parentsString(tp.parents)
         else briefParentsString(tp.parents)
       )
       def isStructuralThisType = (
         // prevents disasters like scala/bug#8158
         owner.isInitialized && owner.isStructuralRefinement && tp == owner.tpe
       )
-      if (isType) typeParamsString(tp) + (
-        if (isClass) " extends " + parents
-        else if (isAliasType) " = " + tp.resultType
-        else tp.resultType match {
-          case rt @ TypeBounds(_, _) => "" + rt
-          case rt                    => " <: " + rt
+      def loop(tp: Type, first: Boolean = false): String = {
```

Comment:
You know what's clever? Shadowing an outer param in an inner loop, so it looks like the whole thing only has one param.

You know what's wrong?